### PR TITLE
feat: Form field creator

### PR DESCRIPTION
**Pull request by FedeMadoery**
# Description

Add a draft Form Field that the creator will use to create the form displayed on the badge type creation, this form will be used by the creator when they create the badge type. This form allows the creator to select a field type, add a name, and a description, and also sort the fields.

Also, this PR adds an eslint config to remove unused imports.

## What's new?

- [x] Add remove un-used imports
- [x] Add FormField Creator
- [x] Add Link to pages on `index.tsx`

## How this PR should be Tested?

- Go to `badge/type/create` and use the form.

# Screenshots (if relevant)

### CODE REVIEW

File: pages/badge/mint/[typeId].tsx [hunk 1]
```diff
@@ -2,11 +2,65 @@ import { ReactElement } from 'react'
...
+  },
+  {
+    label: 'Just a number',
+    description: 'Enter a number',
+    type: KLEROS_LIST_TYPES.NUMBER,
+    isIdentifier: false,
+  },
+  {
+    label: 'Your address',
+    description: 'Enter your eth address',
```
Comment (nicosampler) [inline pages/badge/mint/[typeId].tsx:34]: it will depend on the usage case, perhaps "Enter an eth address" is a better option.

Comment (nicosampler) [reply pages/badge/mint/[typeId].tsx:34]: oh, I see, this is just an example

File: src/components/form/helpers/customSchemas.ts [hunk 2]
```diff
@@ -67,3 +69,8 @@ export const FileSchema = createUniqueFieldSchema(
     .refine(({ file }) => file?.size <= MAX_FILE_SIZE, `Max file size is 5MB.`),
   'FileSchema',
 )
+
+export const FieldTypeSchema = createUniqueFieldSchema(
```
Comment (nicosampler) [inline src/components/form/helpers/customSchemas.ts:73]: can we name it with Kleros? as it is Kleros related

File: src/pagePartials/badgeTypes/FormFieldCreator.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  type: FieldTypeSchema.describe('Field Type // Type of the required data'),
+})
+
+export default function FormFieldCreator({
+  addField,
+  fields,
+  moveField,
+  removeField,
+}: FormFieldSelectorProps) {
+  // Need to type the useForm call accordingly
```
Comment (nicosampler) [inline src/pagePartials/badgeTypes/FormFieldCreator.tsx:35]: is there a way to move all the props inside this component (addFiled, fields, moveField and removeField)?

What we want is that when this component is used inside a form, the key holding the values is updated. Perhaps we can update the instance of the form directly from this component and avoid defining the state logic outside of it.

**Review by nicosampler — COMMENTED** [review]
@FedeMadoery the code looks good. I think we can improve it a little bit to make it simpler to consume. I was thinking we could have something like so:

I'm thinking something like so:



```ts

export const CreateBadgeTypeSchema = z.object({
  name: z.string().describe('Your name // ??'),
  description: z.string().describe('Tell us about you // Tell us about you'),
  // ... other fields
  customData: KlerosDynamicFields.describe('Add the fields the user will need to upload to mint the badge')
 })

// and then...

return {
  <Formfield onSubmit={onSubmit}
        schema={CreateBadgeTypeSchema} />
     }
```

So all the logic of moving fields and the state of the dynamic fields are handled internally.

File: src/pagePartials/badgeTypes/FormFieldItem.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
+import { useRef } from 'react'
```
[the remark this answers is not in the document]
Comment (FedeMadoery) [reply src/pagePartials/badgeTypes/FormFieldItem.tsx:1]: Why? What will be the advantage of doing it?

File: src/components/form/helpers/customSchemas.ts (hunk 2 above)
Comment (FedeMadoery) [reply src/components/form/helpers/customSchemas.ts:73]: Good catch, sure

File: src/pagePartials/badgeTypes/FormFieldItem.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (nicosampler) [reply src/pagePartials/badgeTypes/FormFieldItem.tsx:1]: check the global comment, I mean merge it in a new component that wraps both functionalities. the idea is that when we want to consume it, we don't need to know or be aware of all the internal states the components need. Keep their internal logic encapsulated.

File: src/pagePartials/badgeTypes/FormFieldCreator.tsx (hunk 3 above)
Comment (FedeMadoery) [reply src/pagePartials/badgeTypes/FormFieldCreator.tsx:35]: Yes, for sure, I didn't make it with that approach in mind.

**Comment by FedeMadoery** [comment]
@nicosampler  I made a quick update to accomplish that new requirement. Take a look closely to be sure that makes sense and that the names are consistent, I had to do it quickly and I may make mistakes

**Review by nicosampler — APPROVED** [review]

**Pull request merged**